packages/react-native-adjust/src/withReactNativeAdjust.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  AndroidConfig,
+  createRunOncePlugin,
+  IOSConfig,
+  withXcodeProject,
+  withAppBuildGradle,
+  ConfigPlugin,
+} from "@expo/config-plugins";
+import { mergeContents } from "@expo/config-plugins/build/utils/generateCode";
+
+const withXcodeLinkBinaryWithLibraries: ConfigPlugin<{
+  library: string;
+  status?: string;
+}> = (config, { library, status }) => {
+  return withXcodeProject(config, (config) => {
+    const options = status === "optional" ? { weak: true } : {};
+
+    const target = IOSConfig.XcodeUtils.getApplicationNativeTarget({
+      project: config.modResults,
+      projectName: config.modRequest.projectName!,
+    });
+
+    config.modResults.addFramework(library, {
+      target: target.uuid,
+      ...options,
+    });
+
+    return config;
+  });
+};
+
+export const addAndroidPackagingOptions = (src: string) => {
+  return mergeContents({
+    tag: "react-native-play-services-analytics",
+    src,
+    newSrc: `
+      implementation 'com.google.android.gms:play-services-analytics:18.0.0'
+      implementation 'com.android.installreferrer:installreferrer:2.2'
+    `,
+    anchor: /dependencies(?:\s+)?\{/,
+    // Inside the dependencies block.
+    offset: 1,
+    comment: "//",
+  });
+};
+
+const withGradle: ConfigPlugin = (config) => {
+  return withAppBuildGradle(config, (config) => {
+    if (config.modResults.language === "groovy") {
+      config.modResults.contents = addAndroidPackagingOptions(
+        config.modResults.contents
+      ).contents;
+    } else {
+      throw new Error(
+        "Cannot add Play Services maven gradle because the project build.gradle is not groovy"
+      );
+    }
+    return config;
+  });
+};
+
+/**
+ * Apply react-native-adjust configuration for Expo SDK +44 projects.
+ */
+const withAdjustPlugin: ConfigPlugin = (config) => {
+  config = withXcodeLinkBinaryWithLibraries(config, {
+    library: "iAd.framework",
+    status: "optional",
+  });
+
+  config = withXcodeLinkBinaryWithLibraries(config, {
+    library: "AdServices.framework",
+    status: "optional",
+  });
+
+  config = withXcodeLinkBinaryWithLibraries(config, {
+    library: "AdSupport.framework",
+    status: "optional",
+  });
+
+  config = withXcodeLinkBinaryWithLibraries(config, {
+    library: "StoreKit.framework",
+    status: "optional",
+  });
+
+  config = withXcodeLinkBinaryWithLibraries(config, {
+    library: "AppTrackingTransparency.framework",
+    status: "optional",
+  });
+
+  config = AndroidConfig.Permissions.withPermissions(config, [
+    "com.google.android.gms.permission.AD_ID",
```

Comment:
According to the docs, this is for Android 12 (API 31), Expo currently supports API 30 so you may need to add a plugin to [bump the version](https://github.com/revtel/react-native-nfc-manager/pull/478).

Reply:
Ha yes. The docs says "For apps with target API level set to 30 (Android 11) or older, this permission is not needed.".

Don't you think it's the responsibility of the client application to update their target version?

Reply:
I do but we don't really have a mechanism built in for this yet. The Expo SDK version is tied to a specific iOS and Android API version. Ideally every package would support the same versions and defensively code around versioning issues, but since this isn't always the case we need mechanisms for packages to ensure the runtime is suitable. 

Perhaps a config plugin property could be used to conditionally add the permission and bump the Android API version?

Reply:
That's a neat idea. I'll implement that!